### PR TITLE
feat(plugin): add email composer plugin

### PR DESCRIPTION
**Pull request by cmanou**
First pass at email composer plugin. Should address #12

### CODE REVIEW

File: src/plugins/emailcomposer.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+})
+export class EmailComposer {
+
+  /**
+   * Opens Email Composer with email contents
+   * @param email {email} Email
+   * @returns {Promise<any>} Resolves promise when the EmailComposer has been opened
+   */
+  @Cordova()
+  static open(email: email): Promise<any> { return }
```
Comment (ihadeed) [inline src/plugins/emailcomposer.ts]: Shouldn't it be something like this? 

```js
@Cordova({
successIndex: 1,
errorIndex: 3
})
static open(email: email, scope : any) : Promise<any> {return}
```

File: src/plugins/emailcomposer.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+ * ```
+ */
+@Plugin({
+  plugin: 'cordova-plugin-email-composer',
+  pluginRef: 'emailComposer',
+  repo: 'https://github.com/katzer/cordova-plugin-email-composer.git',
+  platforms: ['Android', 'iOS', 'Windows Phone 8']
+})
+export class EmailComposer {
+
```
Comment (ihadeed) [inline src/plugins/emailcomposer.ts]:
```js
@Cordova()
static isAvailable() : Promise<boolean> {return}
```

Comment (cmanou) [inline src/plugins/emailcomposer.ts]: Missed that when I looked at the ngCordova docs for the plugin will work on adding this. 
Thoughts on whether as first pass we need to implement the overload with the urischeme passed in as well?

**Comment by cmanou** [comment]
Hi All, 
This is working for now though their is an issue with the isAvailable wrapper, since the root call in the cordova plugin the callback "moves" arguments depending on whether arguments are supplied, see [here](https://github.com/katzer/cordova-plugin-email-composer/blob/master/www/email_composer.js#L62)

I don't see an immediate way to do this with the current @Cordova() helper, unless i'm mistaken. What are your opinions on the best way to approach this?

**Comment by ihadeed** [comment]
@cmanou yeah this plugin seems a bit tricky. 

We can wrap the whole function without the `@Cordova` wrapper, something like this:
```js
static isAvailable (app? : string) : Promise<boolean> {
return new Promise<boolean>((res, rej) => {
cordova.plugins.email.isAvailable(app, (isAvailable : boolean) => { if(isAvailable) res(); else rej() });
});
}
```

**Comment by cmanou** [comment]
@ihadeed decided to go the no wrapper way though, kept it as resolving promise boolean as was the original plugin. Think this is now good to go?

**Comment by ihadeed** [comment]
Thanks @cmanou !

**Pull request merged**